packages/jobs/utils/contract-utils.js

```diff
@@ -1,57 +1,79 @@
-const Web3 = require('web3');
-const Promise = require('bluebird');
 const contract = require('truffle-contract');
+const ethers = require('ethers');
 
 const config = require('../config');
 const logger = require('./logger')('utils/contract-utils');
 
-let ContractUtils = {};
-
 const contractsFolder = '../build/contracts/';
-let web3 = null;
+const MAX_ITERATIONS_FOR_WALLET_SEARCHING = 10000;
+const LOG_INTERATIONS_INTERVAL = 1000;
+
+let ContractUtils = {
+  getProvider,
+};
 
-const AliceToken = initializeContract('AliceToken');
-const ClaimsRegistry = initializeContract('ClaimsRegistry');
+ContractUtils.mainWallet = getMainWallet();
+validateNetworkId(ContractUtils.mainWallet.provider);
 
-ContractUtils.getClaimsRegistry = (address) => {
-  return getContractInstance(ClaimsRegistry, address);
+ContractUtils.getTruffleContract = function (name) {
+  const artefacts = require(contractsFolder + name + '.json');
+  return contract(artefacts);
 };
 
-ContractUtils.getProvider = function () {
-  lazyInitWeb3();
-  return web3.currentProvider;
+ContractUtils.getContractInstance = function (
+  contractName,
+  address,
+  wallet=ContractUtils.mainWallet
+) {
+  const contract = ContractUtils.getTruffleContract(contractName);
+  return new ethers.Contract(address, contract.abi, wallet);
 };
 
-ContractUtils.getWeb3 = function () {
-  lazyInitWeb3();
-  validateNetworkId();
-  return web3;
+ContractUtils.getWalletForIndex = function (index) {
+  let path = `m/44'/60'/0'/0/${index}`;
+  return new ethers.Wallet.fromMnemonic(config.mnemonic, path);
 };
 
-Object.assign(ContractUtils, {
-  AliceToken,
+ContractUtils.getWallet = function (address) {
+  for (let counter = 0; counter < MAX_ITERATIONS_FOR_WALLET_SEARCHING; counter++) {
+    if (counter > 0 && counter % LOG_INTERATIONS_INTERVAL == 0) {
+      logger.info(`${counter} wallets checked, but address ${address} still was not found`);
+    }
+    let wallet = ContractUtils.getWalletForIndex(counter);
+    if (equalAddresses(wallet.address, address)) {
+      const provider = getProvider();
+      return wallet.connect(provider);
+    }
+  }
+  throw new Error(`Could not get wallet for address: ${address}`);
+};
 
-  getContractInstance,
-  initializeContract,
-});
+ContractUtils.deployContract = async function (truffleContractObj, ...args) {
+  const { abi, bytecode } = truffleContractObj;
+  const contractFactory = new ethers.ContractFactory(
+    abi,
+    bytecode,
+    ContractUtils.mainWallet);
+  const contract = await contractFactory.deploy(...args);
+  await contract.deployed(); // waiting until it is mined
+  return contract;
+};
 
-function getContractInstance (contract, address) {
-  let contractByAbi = ContractUtils.getWeb3().eth.contract(contract.abi);
-  let contractInstance = contractByAbi.at(address);
-  return Promise.promisifyAll(contractInstance);
-}
+function getProvider() {
+  return new ethers.providers.JsonRpcProvider(config.ethClientAddress);
```

Comment:
It's unclear for me, why do we specify an address in here as we are connecting it to an address-based wallet?

Reply:
ethClientAddress means 'http://localhost:8545' or 'https://rinkeby.infura.io/v3/197cdc79ac3b4198a330490d7cb79f', I will rename it to something more readable like ethEndpointUrl